### PR TITLE
Replace Mongo with Postgres

### PR DESCRIPTION
**Pull request by KludgeKML**
This PR replaces MongoDB in Authenticating Proxy with PostgreSQL.

- Clears long-standing tech debt: AP currently runs on a Mongo 2.6 instance.
- The data in AP isn’t really document-type data, it’s perfectly easy to represent as a relational table.

⚠️ Do not merge until infrastructure PRs (below) are merged and infrastructure created! ⚠️ 

Related PRs:
https://github.com/alphagov/govuk-aws/pull/1640
https://github.com/alphagov/govuk-aws-data/pull/1092
https://github.com/alphagov/govuk-puppet/pull/11967
https://github.com/alphagov/govuk-secrets/pull/1370

# Why
https://trello.com/c/lSpntlfk/81-mongo-26-has-reached-end-of-life

### CODE REVIEW

File: config/database.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  url: <%= ENV["DATABASE_URL"]%>
+
+test:
+  <<: *default
+  database: content_publisher_test
+  url: <%= ENV["TEST_DATABASE_URL"] %>
+
+production:
+  <<: *default
+  database: content_publisher_production
```
Comment (leenagupte) [inline config/database.yml:19]: Copy and paste error?

Comment (KludgeKML) [reply config/database.yml:19]: Yes, definitely.

Comment (KludgeKML) [reply config/database.yml]: Fixed. Thanks!

File: Dockerfile [hunk 2]
```diff
@@ -15,6 +15,8 @@ COPY . /app
 
 
 FROM $base_image
+RUN apt-get update -qq && apt-get upgrade -y
+RUN apt-get install -y libpq-dev && apt-get clean
```
Comment (leenagupte) [inline Dockerfile:19]: @KludgeKML Do you think you might need a puppet entry for postgres in authenticating proxy?

Comment (leenagupte) [reply Dockerfile:19]: e.g. Similar to: https://github.com/alphagov/govuk-puppet/blob/main/modules/govuk/manifests/apps/content_publisher/db.pp and https://github.com/alphagov/govuk-puppet/blob/main/modules/govuk/manifests/apps/content_publisher.pp#L215-L222 to set the env vars?

Comment (KludgeKML) [reply Dockerfile:19]: Ah, yes - I think in between me writing this and me rebasing it a workflow got added, and of course it's Mongo. (and double of course, the govuk reusable actions don't support postgres). Let me see if I can roll my own here.

Comment (KludgeKML) [reply Dockerfile:19]: Ugh. Looks like there's a whole bunch of issues. For the moment none of the apps that use PG have github actions, so I'm going to assume that at some point someone will add it to the reusable action. For now, jenkins only.

**Comment by KludgeKML** [comment]
For future reference, here are some dropped commits wherein I tried to get PG working in github actions:

431a900 Workflow with PG
785013a Lint yaml
1d2d128 fix typo
30a9b45 Add ports, add to run.
71a2c8e Services run on localhost, apparently?
ca42353 Run jobs in container
bf9ec6b Reinstate host name
f8af36f Sudo unnecessary?
10b6ce0 Set ImageOS for container.

File: Dockerfile (hunk 2 above)
Comment (KludgeKML) [reply Dockerfile:19]: This is fixed, github actions working now.

**Review by AgaDufrat — APPROVED** [review]
Tested in all environments 👍 🎉

**Pull request merged**